wordpress-mobile/gutenberg-mobile
# Address Warning that ImageEdit is Using Deprecated componentWillReceiveProps

Related [gutenberg PR](https://github.com/WordPress/gutenberg/pull/16577) 

### Summary

This fixes one of the warnings that we now see when running the demo app because of the use of the [deprecated](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#new-lifecycle-getderivedstatefromprops) `componentWillReceiveProps` component lifecycle method.

On launching the app, you will still see the warning for use of the deprecated `componentWillReceiveProps` method, but now there should be only 1 warning instead of 2 😃 (addressing the other warning is a bit less straightforward, so I wanted to just keep this PR very small and focused).

Before | After
--- | ---
![Screen Shot 2019-07-13 at 1 40 09 PM](https://user-images.githubusercontent.com/4656348/61174889-a4bab680-a574-11e9-8b0b-4580700a718f.png) | ![Screen Shot 2019-07-13 at 1 38 04 PM](https://user-images.githubusercontent.com/4656348/61174891-a8e6d400-a574-11e9-84f5-9dda67a857c8.png)

### Testing

First, check the warnings on loading the demo app and confirm that there is no longer the warning for `ImageEdit`.

Second, verify that the new static method is still addressing the UI flicker that the initial method was added to fix. As noted in the [original PR adding this method](https://github.com/WordPress/gutenberg/pull/14883#discussion_r273769922), this state update was added to avoid a UI flicker in the format bar when switching away from an image caption where the toolbar buttons for both the caption and the newly-selected block were shown at the same time.

1. Open demo app
2. Select an image caption
3. Select another block with rich text formatting
4. Verify that the rich text editing buttons for _both_ the image caption and the newly selected are not briefly both displayed in the format bar (i.e., you see duplicates of each icon).

### Update release notes:
No updates made to release notes because there are no user-facing changes.
